### ISSUE TITLE
Impossible to use for modifications to Aurora Serverless

### ISSUE THREAD

**Issue by marcguyer** [aws-quickstart/quickstart-amazon-aurora-mysql#50]
According to AWS Support:

> Please note that, currently, when you are updating the stack for an Aurora Serverless DB cluster, you can't include changes to any other properties when you specify one of the following properties: PreferredBackupWindow, PreferredMaintenanceWindow, and Port[1].

The template specifies `Port` property which entirely eliminates the utility of the template after initial creation of the instance. The only workaround available at this time appears to be to maintain our own fork of this template with a modification to remove the `Port` property.

I must confess, I don't see a clean way to do this differently for Serverless since I think the fix is to remove the Property altogether. However, I think a complete duplicate of the Resource using a Conditional might be required to support this nuance.

**Comment by tonynv** [comment]
Are you planning on send a PR for this update we happy to review

**Comment by tonynv** [comment]
@davmayd can you please review

**Comment by tonynv** [comment]
> According to AWS Support:
> 
> > Please note that, currently, when you are updating the stack for an Aurora Serverless DB cluster, you can't include changes to any other properties when you specify one of the following properties: PreferredBackupWindow, PreferredMaintenanceWindow, and Port[1].
> 
> The template specifies `Port` property which entirely eliminates the utility of the template after initial creation of the instance. The only workaround available at this time appears to be to maintain our own fork of this template with a modification to remove the `Port` property.
> 
> I must confess, I don't see a clean way to do this differently for Serverless since I think the fix is to remove the Property altogether. However, I think a complete duplicate of the Resource using a Conditional might be required to support this nuance.

Are you planning on send a PR for this update we happy to review

**Comment by davmayd** [comment]
We already have a condition for serverless (https://github.com/aws-quickstart/quickstart-amazon-aurora-mysql/blob/main/templates/aurora_mysql.template.yaml#L709). Is the suggestion here that we duplicate the `AuroraDBCluster` resources, with the duplicate version lacking the `port` parameter? If so, @marcguyer do you plan on submitting a pull request?

**Comment by marcguyer** [comment]
>  Is the suggestion here that we duplicate the AuroraDBCluster resources, with the duplicate version lacking the port parameter?

Yeah, that was my thought but it seems crazy. Is there a better way? I haven't tested this but I wonder if using `AWS::NoValue` for the `Port` parameter in this case would be a successful workaround...

**Comment by marcguyer** [comment]
> I haven't tested this but I wonder if using AWS::NoValue for the Port parameter in this case would be a successful workaround...

Tested without success. Our workaround is clean and I don't see a good solution for this here without an enormous amount of complexity with very little benefit. 

> If so, @marcguyer do you plan on submitting a pull request?

Sorry, no, not at this time.
